wire-ios/Wire-iOS/Sources/Helpers/SaveFileManager.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import UIKit
+import WireSystem
+
+protocol SaveFileActions {
+    func save(value: String, fileName: String, type: String)
+}
+
+final class SaveFileManager: NSObject, SaveFileActions {
+    private var pendingSaveURLs = [URL]()
+
+    private let logger: LoggerProtocol
+    private let systemSaveFilePresenter: SystemSaveFilePresenting
+
+    public init(systemFileSavePresenter: SystemSaveFilePresenting, logger: LoggerProtocol) {
+        self.systemSaveFilePresenter = systemFileSavePresenter
+        self.logger = logger
+    }
+
+    func save(value: String, fileName: String, type: String) {
+        let temporaryDirectoryURL = URL(fileURLWithPath: NSTemporaryDirectory(), isDirectory: true)
+        let fileURL = temporaryDirectoryURL.appendingPathComponent(fileName + "." + type)
```

Comment:
We have an extension `URL.directoryURL(...)` than can be reused.

Reply:
We cannot use it because we have to use the temporary directory to store this file within the app sandbox.